### PR TITLE
modify envs api names

### PR DESCRIPTION
**Pull request by giteden**
## Proposed Changes

new names for 'compose', 'merge', and 'override' methods and/or their args.

### CODE REVIEW

File: scopes/envs/envs/environments.main.runtime.ts [hunk 1]
```diff
@@ -92,6 +92,19 @@ export class EnvsMain {
   }
 
   /**
+   * apply a list of Env Transformers on an Env, to produce a new (transformed) Env
+   */
+  transformEnv(baseEnv: Environment, envTransformers: EnvTransformer[]) {
```
Comment (GiladShoham) [inline scopes/envs/envs/environments.main.runtime.ts:97]: change to apply env
change base env back to target env

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/envs/envs/environments.main.runtime.ts [hunk 2]
```diff
@@ -113,28 +137,28 @@ export class EnvsMain {
   }
 
   /**
-   * compose two environments into one.
+   * merge two environments into one.
    */
-  merge<T>(targetEnv: Environment, sourceEnv: Environment): T {
+  merge<T>(transformationEnv: Environment, baseEnv: Environment): T {
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/envs/envs/environments.main.runtime.ts]: changed back to target and sources

File: scopes/envs/envs/environments.main.runtime.ts [hunk 3]
```diff
@@ -113,28 +137,28 @@ export class EnvsMain {
   }
 
   /**
-   * compose two environments into one.
+   * merge two environments into one.
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/envs/envs/environments.main.runtime.ts]: done. didn't mention item 3. seems to go too much into the implementation.

File: scopes/envs/envs/environments.main.runtime.ts [hunk 4]
```diff
@@ -104,6 +117,17 @@ export class EnvsMain {
   }
 
   /**
+   * create an Env Transformer out of an Env.
+   * (an Env Transformer modifies or adds specific properties to an Env)
+   */
+  createEnvTransformer(transformationEnv: Environment): EnvTransformer {
+    return (env: Environment) => {
+      return this.merge(transformationEnv, env);
+    };
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/envs/envs/environments.main.runtime.ts]: override is depercated

File: scopes/envs/envs/environments.main.runtime.ts [hunk 5]
```diff
@@ -104,6 +117,17 @@ export class EnvsMain {
   }
 
   /**
+   * create an Env Transformer out of an Env.
+   * (an Env Transformer modifies or adds specific properties to an Env)
+   */
+  createEnvTransformer(transformationEnv: Environment): EnvTransformer {
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/envs/envs/environments.main.runtime.ts]: used 'envProps'

File: scopes/envs/envs/environments.main.runtime.ts (hunk 1 above)
Comment (giteden) [reply scopes/envs/envs/environments.main.runtime.ts]: done

**Comment by GiladShoham** [comment]
not relevant anymore

**Pull request closed without merging**